### PR TITLE
[Tests] Properly remove exported files

### PR DESCRIPTION
**Pull request by alxbilger**
Based on https://github.com/sofa-framework/sofa/pull/3961

Some exported files were not removed after executing unit tests.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: Sofa/Component/IO/Mesh/tests/MeshExporter_test.cpp [hunk 1]
```diff
@@ -68,142 +79,181 @@ class MeshExporter_test : public BaseSimulationTest,
...
-#define NUM_PARAMS (unsigned int)2
-
 /// run the tests
-TEST_P( MeshExporter_test, checkBasicBehavior) {
-    std::vector<string> params = GetParam() ;
-    ASSERT_EQ(params.size(), NUM_PARAMS );
-    ASSERT_NO_THROW( this->checkBasicBehavior(params, "outfile", {"outfile."+params[0]}) ) ;
+TEST_P(MeshExporter_test, checkBasicBehavior)
+{
+    const TestParam& params = GetParam();
```
Comment (epernod) [inline Sofa/Component/IO/Mesh/tests/MeshExporter_test.cpp:172]: you removed the test: ASSERT_EQ(params.size(), NUM_PARAMS );
is it on purpose?

File: Sofa/Component/IO/Mesh/tests/MeshExporter_test.cpp [hunk 2]
```diff
@@ -68,142 +79,181 @@ class MeshExporter_test : public BaseSimulationTest,
...
+    std::vector<std::string> paths;
+    for (unsigned int i = 0; i < 20; ++i)
+    {
+        std::stringstream ss;
+        ss << std::setw(5) << std::setfill('0') << i;
+
+        paths.push_back( tempdir + "/" + std::string(filename) + ss.str() + "." + params.extension);
+        for (const auto& addExtension : params.additionalExtensions)
+        {
+            paths.push_back(tempdir + "/" + std::string(filename)+ ss.str() + "." + addExtension);
```
Comment (epernod) [inline Sofa/Component/IO/Mesh/tests/MeshExporter_test.cpp]: I don't get where the local variable `paths ` is used in this test?

**Comment by epernod** [comment]
As it involve removing/deleting things during test phase, would be good to wait for mac to be back and run all tests on the ci.

File: Sofa/Component/IO/Mesh/tests/MeshExporter_test.cpp (hunk 1 above)
Comment (alxbilger) [reply Sofa/Component/IO/Mesh/tests/MeshExporter_test.cpp:172]: Yes, it is on purpose. Before, the data structure was a `std::vector<string>` (a dynamic container that can have any size), after it is a `struct TestParam`, so not really a dynamic container.

File: Sofa/Component/IO/Mesh/tests/MeshExporter_test.cpp (hunk 2 above)
Comment (alxbilger) [reply Sofa/Component/IO/Mesh/tests/MeshExporter_test.cpp]: Yes, good catch. I'll fix it

**Comment by alxbilger** [comment]
[ci-build][with-all-tests]

**Pull request merged**